### PR TITLE
new docs.sourcegraph.com design

### PR DESCRIPTION
**Pull request by sqs**
Goals:

- Improve/modernize look-and-feel
- Simplify sidebar content, implementation, and how-to-update process
- Add dark theme

# Old sidebar

image

# New sidebar

image
image
image

(Responsive and dark theme screenshots also shown)

---

Relevant Slack discussions:

- Initial: https://sourcegraph.slack.com/archives/C07KZF47K/p1598388973084000?thread_ts=1598295302.057500&cid=C07KZF47K
- Recent: https://sourcegraph.slack.com/archives/C0HMGV90V/p1601510298050600

### CODE REVIEW

File: doc/_resources/assets/docsite.css [hunk 1]
```diff
@@ -0,0 +1,765 @@
...
+}
+body > #sidebar #search-form .search-icon:after {
+    content: "\26B2";
+    font-size: 1.5rem;
+    color: var(--link-color);
+    transform: rotate(-45deg);
+    display: inline-block;
+    position: absolute;
+    left: 28px;
+    top: 13px;
```
Comment (mrnugget) [inline doc/_resources/assets/docsite.css:235]: With `13px` it looks to me like this:

image

With `9px` it looks better:

image


macOS, Chrome Version 85.0.4183.121

Comment (sqs) [reply doc/_resources/assets/docsite.css:235]: Ah, bummer. For me `9px` looks weird:

image

I'll have to come up with another way to do this. Thanks for the catch.

Comment (sqs) [reply doc/_resources/assets/docsite.css:235]: using svg instead, different OSes will use different fonts for this, thanks to https://www.figma.com/blog/when-fonts-fall/ for making this top of mind for me!

**Comment by sqs** [comment]
Based on the positive feedback, I'm going to merge this. Please keep the feedback coming to me and @christinaforney!

**Pull request merged**